### DISCUSSION TITLE
Need for a UID for panels in dashboards

### DISCUSSION THREAD

**Discussion by ali-avci** [grafana/grafana#57740] [Alerting]
While developing the alert provisioning API, I've seen a IMO a weak point in the design of the reference between an alert and the panel that is associated to that alert.

I've had this situation, after having the dashboards created and the required alerts via the provisioning api, I was asked to add a new panel to the dashboard in a given position. In order not to get in trouble I first deleted all alerts associated with that dashboard and re-created the alerts from scratch. 

The ID of a panel is generated automatically during the post of the dashboard, hence the reference might be invalid after that re-indexing did occur in the panels. 

```
# sample snippet from the alert json

  "annotations": {
    "__dashboardUid__": "user defined uid",
    "__panelId__": "2"   # reference to a regenerated ID 
  },
```

Since the relationship between the dashboard and the panel id is already handled the UID needs to be unique only within the dashboard object.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
